include/pushmi/o/async.h

```diff
@@ -227,45 +223,40 @@ namespace detail {
   };
 
   // Generalisation to customise the entire protocol
-  // TODO: The In template parameter here makes this customisation point
-  // seem strange
-  template<class In, class Executor, class Out>
+  template<class MakeReceiver, class Executor, class Out>
   auto async_fork_customization(Executor exec, Out out) {
-    return async_fork_customization_generic<Executor, In>{}(out, exec);
+    return async_fork_customization_generic<Executor, MakeReceiver>{}(out, exec);
```

Comment:
The overall goal is to work out what the type of our customisation point needs to be. This may change a little when I hoist this up a level - I think we actually want to make the sender not the receiver because what we will get back from an executor will be a sender.

Reply:
I agree this change is only a small improvement from the API-design PoV, and not the final shape the customization point should have. This change is more about eliminating needless compile-time dependencies.